exporter/alertmanagerexporter/alertmanager_exporter.go

```diff
@@ -23,20 +33,172 @@ type alertmanagerExporter struct {
+type alertmanagerEvent struct {
+	spanEvent ptrace.SpanEvent
+	traceID   string
+	spanID    string
+	severity  string
+}
+
+func (s *alertmanagerExporter) convertEventSliceToArray(eventSlice ptrace.SpanEventSlice, traceID pcommon.TraceID, spanID pcommon.SpanID) []*alertmanagerEvent {
+	if eventSlice.Len() > 0 {
+		events := make([]*alertmanagerEvent, eventSlice.Len())
 
-	// To Be Implemented
+		for i := 0; i < eventSlice.Len(); i++ {
+			var severity string
+			severityAttrValue, ok := eventSlice.At(i).Attributes().Get(s.severityAttribute)
+			if ok {
+				severity = severityAttrValue.AsString()
+			} else {
+				severity = s.defaultSeverity
+			}
+			event := alertmanagerEvent{
+				spanEvent: eventSlice.At(i),
+				traceID:   traceID.String(),
+				spanID:    spanID.String(),
+				severity:  severity,
+			}
+
+			events[i] = &event
+		}
+		return events
+	}
 	return nil
 }
 
-func (s *alertmanagerExporter) start(_ context.Context, _ component.Host) error {
+func (s *alertmanagerExporter) extractEvents(td ptrace.Traces) []*alertmanagerEvent {
+
+	// Stitch parent trace ID and span ID
+	rss := td.ResourceSpans()
+	var events []*alertmanagerEvent
+	if rss.Len() == 0 {
+		return nil
+	}
+
+	for i := 0; i < rss.Len(); i++ {
+		resource := rss.At(i).Resource()
+		ilss := rss.At(i).ScopeSpans()
+
+		if resource.Attributes().Len() == 0 && ilss.Len() == 0 {
+			return nil
+		}
+
+		for j := 0; j < ilss.Len(); j++ {
+			spans := ilss.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				traceID := spans.At(k).TraceID()
+				spanID := spans.At(k).SpanID()
+				events = append(events, s.convertEventSliceToArray(spans.At(k).Events(), traceID, spanID)...)
+			}
+		}
+	}
+	return events
+}
+
+func createAnnotations(event *alertmanagerEvent) model.LabelSet {
+	labelMap := make(model.LabelSet, event.spanEvent.Attributes().Len()+2)
+	event.spanEvent.Attributes().Range(func(key string, attr pcommon.Value) bool {
+		labelMap[model.LabelName(key)] = model.LabelValue(attr.AsString())
+		return true
+	})
+	labelMap["TraceID"] = model.LabelValue(event.traceID)
+	labelMap["SpanID"] = model.LabelValue(event.spanID)
+	return labelMap
+}
+
+func (s *alertmanagerExporter) convertEventsToAlertPayload(events []*alertmanagerEvent) []model.Alert {
+
+	payload := make([]model.Alert, len(events))
+
+	for i, event := range events {
+		annotations := createAnnotations(event)
+
+		alert := model.Alert{
+			StartsAt:     time.Now(),
+			Labels:       model.LabelSet{"severity": model.LabelValue(event.severity), "event_name": model.LabelValue(event.spanEvent.Name())},
+			Annotations:  annotations,
+			GeneratorURL: s.generatorURL,
+		}
+
+		payload[i] = alert
+	}
+	return payload
+}
+
+func (s *alertmanagerExporter) postAlert(ctx context.Context, payload []model.Alert) error {
+
+	msg, err := json.Marshal(payload)
+	if err != nil {
+		s.settings.Logger.Debug("error marshaling alert to JSON", zap.Error(err))
```

Comment:
When you are returning an error, avoid logging it. Otherwise, users will end up seeing the same message twice: here, and from the caller.

If you are NOT returning an error (ie, when the error isn't decisive to stop the processing), it's fine to log it.